feat(user-guide): Spring 6.0 and Spring Boot 3.1 support (#1486)
camunda/camunda-docs-manual: content/update/minor/719-to-720/_index.md

[quarkus3]: https://quarkus.io/blog/quarkus-3-0-final-released
[quarkus3-update]: https://quarkus.io/blog/quarkus-3-0-final-released/#upgrading
[quarkus-integration]: {{< ref "/user-guide/quarkus-integration" >}}

# Spring Framework 6.0 support

> The newly created `engine-spring-6` module provides support to Spring Framework 6.0 with maven coordinates:

Reviewer: ```suggestion
The newly created `engine-spring-6` module provides support to Spring Framework 6.0 with the following maven coordinates:
```

```xml
<dependency>
  <groupId>org.camunda.bpm</groupId>
  <artifactId>camunda-engine-spring-6</artifactId>
</dependency>
```

The most noteworthy changes from the new major version are JDK 17+ and Jakarta EE 9+ baseline.

> In order to adjust your applications, follow the Spring Framework [upgrade guide][spring6-guide]

Reviewer: ```suggestion
To adjust your applications, follow the Spring Framework [upgrade guide][spring6-guide]
```

> and check [Spring Framework 6.0 goes GA][spring6] blog post.

Reviewer: ```suggestion
and check the [Spring Framework 6.0 goes GA][spring6] blog post.
```

Camunda Platform introduces Spring Framework 6.0 support for:

* Spring Boot Starter
* WildFly Application Server 27

[spring6]: https://spring.io/blog/2022/11/16/spring-framework-6-0-goes-ga
[spring6-guide]: https://github.com/spring-projects/spring-framework/wiki/Upgrading-to-Spring-Framework-6.x

# Upgrade to Spring Boot 3.1

> The Camunda Engine now offers support for Spring Boot 3.1. The new major is build on Spring Framework 6.0 

Reviewer: ```suggestion
The Camunda Engine now offers support for Spring Boot 3.1. The new major version builds on Spring Framework 6.0 
```

> and brings changes such as JDK 17 baseline and switch to Jakarta EE 10.

Reviewer: ```suggestion
and brings changes such as JDK 17 baseline and switching to the Jakarta namespace.
```

> For complete list of new features and changes, check the [Spring Boot 3.0][boot30] and [Spring Boot 3.1][boot31] release notes.

Reviewer: ```suggestion
For a complete list of new features and changes, check the [Spring Boot 3.0][boot30] and [Spring Boot 3.1][boot31] release notes.
```

> Have a look the Spring Boot [update guide][boot30-guide] to migrate your applications from Spring Boot 2.7.

Reviewer: ```suggestion
Have a look at the Spring Boot [update guide][boot30-guide] to migrate your applications from Spring Boot 2.7.
```

```xml
<dependency>
  <groupId>org.camunda.bpm</groupId>
  <artifactId>camunda-engine-spring-6</artifactId>
</dependency>
```

The most noteworthy changes from the new major version are JDK 17+ and Jakarta EE 9+ baseline.
To adjust your applications, follow the Spring Framework [upgrade guide][spring6-guide]
and check the [Spring Framework 6.0 goes GA][spring6] blog post.

Camunda Platform introduces Spring Framework 6.0 support for:

* Spring Boot Starter
* WildFly Application Server 27

[spring6]: https://spring.io/blog/2022/11/16/spring-framework-6-0-goes-ga
[spring6-guide]: https://github.com/spring-projects/spring-framework/wiki/Upgrading-to-Spring-Framework-6.x

# Upgrade to Spring Boot 3.1

The Camunda Engine now offers support for Spring Boot 3.1. The new major version builds on Spring Framework 6.0 
and brings changes such as JDK 17 baseline and switching to the Jakarta namespace. 

For a complete list of new features and changes, check the [Spring Boot 3.0][boot30] and [Spring Boot 3.1][boot31] release notes.
Have a look at the Spring Boot [update guide][boot30-guide] to migrate your applications from Spring Boot 2.7.

> Among this changes Spring Boot Starter now relies on Camunda Platform modules that are created for the Jakarta namespace.
> For example, in case you want to build custom webjar, keep in mind that now:

Reviewer: ```suggestion
The switch to the Jakarta namespace requires the Spring Boot Starter and other modules to now rely on Camunda Platform modules that are created for this purpose.
For example, if you want to build a custom webjar, keep the following changes in mind:
```

* `camunda-webapp-webjar` depends on `camunda-webapp-jakarta` for building the Spring Boot webjar.

> * `camunda-webapp-webjar-ee` depends on `camunda-webapp-jakarta` and `camunda-webapp-ee-plugins-jakarta`

Reviewer: ```suggestion
* `camunda-webapp-webjar-ee` depends on `camunda-webapp-jakarta` and `camunda-webapp-ee-plugins-jakarta`.
```
